app/Foliage/CmdLint.hs

```diff
@@ -0,0 +1,46 @@
+module Foliage.CmdLint (cmdLint, lintOptionsParser, LintOptions) where
+
+import Control.Monad (void, when)
+import Development.Shake
+import Foliage.Meta (readPackageVersionSpec, writePackageVersionSpec)
+import Options.Applicative
+import System.FilePath ((</>))
+
+newtype LintOptions = LintOptions {lintOptsInputDir :: FilePath}
+
+lintOptionsParser :: Parser LintOptions
+lintOptionsParser =
+  LintOptions
+    <$> strOption
+      ( long "input-directory"
+          <> metavar "INPUT"
+          <> help "Repository input directory"
+          <> showDefault
+          <> value "_sources"
+      )
+
+cmdLint :: LintOptions -> IO ()
+cmdLint buildOptions = do
+  shake opts $
+    do
+      phony "lintAction" (lintAction buildOptions)
+      want ["lintAction"]
+  where
+    cacheDir = "_cache"
+    opts = shakeOptions {shakeFiles = cacheDir, shakeVerbosity = Verbose}
```

Comment:
Should we share this stuff? Seems like the shake options should pretty much always be the same?

Reply:
We should. Also we should wire the verbosity to a command line flag. Also I don't like how shake prints messages.